app/src/main/java/com/example/ohthmhyh/entities/User.java

```diff
@@ -52,47 +61,126 @@ public void setUsername(String username) {
 
 
     /**
-     * Set the picture ID counter for this user
-     * @param UPIDCounter the number to set the counter to
+     * update the users username. Check if the username has been taken first!
      */
-    public void setUPIDCounter(int UPIDCounter) {
-        this.UPIDCounter = UPIDCounter;
+    public void updateUsername(String username) {
+        this.username = username;
+        dba.pushUser(this);
     }
 
 
     /**
-     * Get the picture ID counter for this user
-     * @return  the value of the UPID counter
+     * Set the users friend list as a list of UID's
+     * @param friends  The users friend list as UID's
      */
-    public int getUPIDCounter() {
-        return UPIDCounter;
+    public void setFriendList(ArrayList<String> friends){
+        this.friendList = friends;
     }
 
 
     /**
-     * Get the next available picture id for this user (auto-increments)
-     * @return  the next available picture id for the user
+     * Get the users friend list as a list of UID's
+     * @return  The users friend list as UID's
      */
-    public int nextUPID() {
-        return UPIDCounter++;
+    public ArrayList<String> getFriendList(){
+        return friendList;
     }
 
 
     /**
-     * Set the users friend list as a list of UID's
-     * @param friends  The users friend list as UID's
+     * Set the users incoming friend request list as a list of UID's
+     * @param friendRequests  The users friend request list as UID's
      */
-    public void setFriendList(ArrayList<String> friends){
-        this.friendList = friends;
+    public void setFriendRequests(ArrayList<String> friendRequests){
+        this.friendRequests = friendRequests;
     }
 
 
     /**
-     * Get the users friend list as a list of UID's
-     * @return  The users friend list as UID's
+     * Get the users incoming friend request list as a list of UID's
+     * @return  The users friend request list as UID's
      */
-    public ArrayList<String> getFriendList(){
-        return friendList;
+    public ArrayList<String> getFriendRequests(){
+        return friendRequests;
+    }
+
+
+    /**
+     * Sends a friend request to the person who currently has the username. Make sure the
+     * username is valid first!
+     */
+    public void sendFriendRequest(String username){
+        // make sure we aren't sending a friend request to ourself
+        if(username == this.username){
```

Comment:
Oh. I used the wrong comparison for strings. That's why you can add yourself

Reply:
Oh okay good catch. Is it also possible to detect if it's a user that has already been added?

Reply:
Done. Not sure how I didn't catch this stuff yesterday...

Reply:
Awesome thanks so much, will review right now.